### PR TITLE
Avoid linking to itself in implementors section of trait page

### LINKED ISSUE #32474: rustdoc should avoid unnecessary links into the containing page itself
For example,

- http://doc.rust-lang.org/std/iter/trait.IntoIterator.html#implementors

`IntoIterator` linked to `trait.IntoIterator.html` are only distracting. By the nature of the "implementer" section, they refer to the containing page itself.

### PR DESCRIPTION
**Pull request by sanxiyn**
Fix #32474.

### CODE REVIEW

File: src/librustdoc/html/format.rs [hunk 1]
```diff
@@ -574,6 +574,20 @@ impl fmt::Display for clean::Impl {
...
+    pub fn fmt_for_trait_page(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "impl{} ", self.generics)?;
+        if let Some(ref ty) = self.trait_ {
+            write!(f, "{}{} for ",
+                   if self.polarity == Some(clean::ImplPolarity::Negative) { "!" } else { "" },
+                   ty.trait_name().unwrap())?;
+        }
+        write!(f, "{}{}", self.for_, WhereClause(&self.generics))?;
+        Ok(())
+    }
```
Comment (alexcrichton) [inline src/librustdoc/html/format.rs]: Since this is just duplicating the code above, can the implementations be refactored?

**Comment by sanxiyn** [comment]
Extracted duplicated code to a function.

**Comment by alexcrichton** [comment]
Thanks! Can you add a test for this as well?

**Comment by sanxiyn** [comment]
Added a test.

**Comment by sanxiyn** [comment]
Ping.

**Comment by steveklabnik** [comment]
Gah! Sorry this slipped by.

@bors: r+ rollup

**Pull request merged**